proselint/registry/__init__.py

```diff
@@ -54,19 +71,23 @@ def get_all_enabled(
         self, enabled: dict[str, bool] = DEFAULT["checks"]
     ) -> list[Check]:
         """Filter registered checks by config values based on their keys."""
-        self.enabled_checks = enabled
-
-        enabled_checks: list[str] = []
-        skipped_checks: list[str] = []
-        for key, key_enabled in self.enabled_checks.items():
-            (skipped_checks, enabled_checks)[key_enabled].append(key)
+        self.enabled_checks = _flatten_config(enabled)
+        by_specificity = sorted(
+            self.enabled_checks.items(),
+            key=lambda x: x[0].count("."),
+            reverse=True,
+        )
```

Comment:
The configuration being normalised (i.e. flattened) should be a stipulation of this function for performance reasons. It may also be worth requiring the input configuration to be sorted in this way, which can be applied as a post-merge transform like `_flatten_config`.